modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/stat/IgniteStatisticsManagerImpl.java

```diff
@@ -124,6 +137,52 @@ else if (db == null)
             mgmtPool,
             ctx::log
         );
+
+        ctx.internalSubscriptionProcessor().registerDistributedConfigurationListener(dispatcher -> {
+            usageState.addListener((name, oldVal, newVal) -> {
+                if (log.isInfoEnabled())
+                    log.info(String.format("Statistics usage state was changed from %s to %s", oldVal, newVal));
+
+                if (oldVal == newVal)
+                    return;
+
+                switch (newVal) {
+                    case OFF:
+                        disableOperations();
+
+                        break;
+                    case ON:
+                    case NO_UPDATE:
+                        enableOperations();
+
+                        break;
+                }
+            });
+
+            dispatcher.registerProperty(usageState);
+        });
+
+        StatisticsUsageState currState = usageState();
+        if (currState == ON || currState == NO_UPDATE)
+            enableOperations();
+    }
+
+    /**
+     * Enable statistics operations.
+     */
+    private void enableOperations() {
+
```

Comment:
empty line

Reply:
Is it possible enableOperations() and disableOperations() is called out of order?
Does it make sense to synchronize them?

Reply:
Has sense - I'll add synchronized to both of them.